evalml/automl/automl_algorithm/automl_algorithm.py

```diff
@@ -64,14 +65,72 @@ def next_batch(self):
+    def _set_allowed_pipelines(self, allowed_pipelines):
+        """Sets the allowed parameters and creates the tuners for the input pipelines."""
+        self.allowed_pipelines = allowed_pipelines
+        for pipeline in self.allowed_pipelines:
+            self._create_tuner(pipeline)
+
+    def _create_tuner(self, pipeline):
+        """Creates a tuner given the input pipeline."""
+        pipeline_hyperparameters = pipeline.get_hyperparameter_ranges(
+            self._hyperparameters
+        )
+        self._tuners[pipeline.name] = self._tuner_class(
+            pipeline_hyperparameters, random_seed=self.random_seed
+        )
+
+    def _separate_hyperparameters_from_parameters(self):
+        """Seperate out the parameter and hyperparameter values from the search parameters dict."""
+        for key, value in self.search_parameters.items():
+            hyperparam = {}
+            param = {}
+            for name, parameters in value.items():
+                if isinstance(parameters, (Integer, Categorical, Real)):
+                    hyperparam[name] = parameters
+                else:
+                    param[name] = parameters
+            if hyperparam:
+                self._hyperparameters[key] = hyperparam
+            if param:
+                self._pipeline_parameters[key] = param
+
     def _transform_parameters(self, pipeline, proposed_parameters):
-        """Given a pipeline parameters dict, make sure pipeline_params, custom_hyperparameters, n_jobs are set properly.
+        """Given a pipeline parameters dict, make sure pipeline_parameters, custom_hyperparameters, n_jobs are set properly.
 
         Arguments:
             pipeline (PipelineBase): The pipeline object to update the parameters.
             proposed_parameters (dict): Parameters to use when updating the pipeline.
         """
+        parameters = {}
+        if "pipeline" in self._pipeline_parameters:
+            parameters["pipeline"] = self._pipeline_parameters["pipeline"]
+        for (
+            name,
+            component_instance,
+        ) in pipeline.component_graph.component_instances.items():
+            component_class = type(component_instance)
+            component_parameters = proposed_parameters.get(name, {})
+            init_params = inspect.signature(component_class.__init__).parameters
+            # Only overwrite the parameters that were passed in on pipeline parameters
+            # if they don't exist in the propsed parameters
+            if name in self._pipeline_parameters and name not in component_parameters:
+                for param_name, value in self._pipeline_parameters[name].items():
+                    component_parameters[param_name] = value
+            # Inspects each component and adds the following parameters when needed
+            if "n_jobs" in init_params:
+                component_parameters["n_jobs"] = self.n_jobs
+            try:
+                if "number_features" in init_params:
```

Comment:
Will file an issue to delete this. Don't know what it's used for and why it's only for iterative algorithm.